cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_service_manager.go

```diff
@@ -218,6 +221,29 @@ func (csm *cloudServiceManager) GetInstances(groupID string) ([]cloudprovider.In
 	return instances, nil
 }
 
+func (csm *cloudServiceManager) ScaleDownCluster(groupID string, instanceIds []string) error {
+	asClient := csm.getASClientFunc()
+
+	instanceDelete := "yes"
+	opts := &huaweicloudsdkasmodel.UpdateScalingGroupInstanceRequest{
+		ScalingGroupId: groupID,
+		Body: &huaweicloudsdkasmodel.UpdateScalingGroupInstanceRequestBody{
+			InstancesId:    instanceIds,
+			InstanceDelete: &instanceDelete,
+			Action:         huaweicloudsdkasmodel.GetUpdateScalingGroupInstanceRequestBodyActionEnum().REMOVE,
+		},
+	}
+
+	_, err := asClient.UpdateScalingGroupInstance(opts)
```

Comment:
The reason you don't use [DeleteScalingInstance](https://github.com/huaweicloud/huaweicloud-sdk-go-v3/blob/8733c595be33e3f957c7532bfa0c091fc2731571/services/as/v1/as_client.go#L143) because it used for delete `one` instance in each call, right?

Reply:
for Delete Scaling Instance it will only delete those instances but won't modify desire instance number of auto scaling group, so that why I'm using this API

Reply:
That's ok.